src/Engine/Trigger/TriggerRepeat.cpp

```diff
@@ -0,0 +1,218 @@
+#include "TriggerRepeat.hpp"
+
+#include "../Component/RigidBody.hpp"
+#include "../Component/Shape.hpp"
+#include "../Entity/Entity.hpp"
+#include "../Manager/Managers.hpp"
+#include "../Manager/PhysicsManager.hpp"
+#include "../Manager/ScriptManager.hpp"
+#include "../Physics/Shape.hpp"
+#include "../Hymn.hpp"
+
+TriggerRepeat::TriggerRepeat() {
+
+}
+
+TriggerRepeat::~TriggerRepeat() {
+
+}
+
+Entity* GetEntity(int GUID) {
+
+    const std::vector<Entity*> entities = Hymn().world.GetEntities();
+    for (int i = 0; i < entities.size(); i++) {
+
+        if (entities[i]->GetUniqueIdentifier() == GUID) {
+
+            return entities[i];
+        }
+    }
+
+    return nullptr;
+}
+
+// ADD SUPPORT FOR VECTOR
+void TriggerRepeat::OnEnter() {
+    if (!collidedEntity.empty() && collidedEntity.front()->GetComponent<Component::RigidBody>()) {
+        Component::RigidBody* rigidBodyComp = collidedEntity.front()->GetComponent<Component::RigidBody>();
+        Managers().physicsManager->ForgetTriggerEnter(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerRetain(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerLeave(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->OnTriggerEnter(triggerVolume, rigidBodyComp, std::bind(&TriggerRepeat::HandleTriggerEvent, this));
+    }
+}
+
+// ADD SUPPORT FOR VECTOR
+void TriggerRepeat::OnRetain() {
+    if (!collidedEntity.empty() && collidedEntity.front()->GetComponent<Component::RigidBody>()) {
+        Component::RigidBody* rigidBodyComp = collidedEntity.front()->GetComponent<Component::RigidBody>();
+        Managers().physicsManager->ForgetTriggerEnter(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerRetain(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerLeave(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->OnTriggerRetain(triggerVolume, rigidBodyComp, std::bind(&TriggerRepeat::HandleTriggerEvent, this));
+    }
+}
+
+// ADD SUPPORT FOR VECTOR
+void TriggerRepeat::OnLeave() {
+
+    if (!collidedEntity.empty() && collidedEntity.front()->GetComponent<Component::RigidBody>()) {
+        Component::RigidBody* rigidBodyComp = collidedEntity.front()->GetComponent<Component::RigidBody>();
+        Managers().physicsManager->ForgetTriggerEnter(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerRetain(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->ForgetTriggerLeave(triggerVolume, rigidBodyComp);
+        Managers().physicsManager->OnTriggerLeave(triggerVolume, rigidBodyComp, std::bind(&TriggerRepeat::HandleTriggerEvent, this));
+    }
+}
+
+std::string TriggerRepeat::GetName() {
+    return name;
+}
+
+void TriggerRepeat::SetName(std::string value) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Function parameter 'value' should be passed by reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3ApassedByValue)